lib/elixir/lib/macro.ex

```diff
@@ -131,7 +131,55 @@ defmodule Macro do
 
   @typep output_expr :: {output_expr | atom, metadata, atom | [output]}
 
-  @typedoc "A keyword list of AST metadata"
+  @typedoc """
+  A keyword list of AST metadata.
+
+  The metadata in Elixir AST is a keyword list of values. Any key can be used
+  and different parts of the compiler may use different keys. For example,
+  the AST received by a macro will always include the `:line` annotation,
+  while the AST emitted by a quote never has the `:line` annotation.
```

Comment:
The line can be provided as option in `quote`:

```elixir
iex(1)> quote line: 3 do
...(1)> 1 + 2
...(1)> end
{:+, [line: 3, context: Elixir, import: Kernel], [1, 2]}
```

Reply:
@fertapric How about this?

```
The metadata in Elixir AST is a keyword list of values. Any key can be used
and different parts of the compiler may use different keys. For example,
the AST received by a macro will always include the `:line` annotation,
while the AST emitted by a `quote/2` only has the `:line` annotation if the
`line` argument was provided to `quote/2`.
```